OrbitGl/App.cpp

```diff
@@ -163,11 +163,39 @@ void OrbitApp::OnCaptureComplete() {
         if (capture_stopped_callback_) {
           capture_stopped_callback_();
         }
+        if (open_capture_finished_callback_) {
+          open_capture_finished_callback_();
+        }
 
         FireRefreshCallbacks();
       });
 }
 
+void OrbitApp::OnCaptureCancelled() {
+  main_thread_executor_->Schedule([this]() mutable {
+    if (capture_failed_callback_) {
+      capture_failed_callback_();
+    }
+    if (open_capture_failed_callback_) {
```

Comment:
I guess the result is correct as the `open_capture_failed_callback_` is effectively just `loading_capture_dialog->close();`, but conceptually it's confusing that both `capture_failed_callback_` and `open_capture_failed_callback_` are called here (they are always set), regardless of whether the capture is being taken or it's being loaded.
Same in `OnCaptureFailed` at line 191 and similarly in `OnCaptureComplete` at line 166 with `open_capture_finished_callback_`.

Reply:
That was part of the idea of combining the handling of loading/capturing. I admit, that it is not ideal yet. Long term it should be fine as soon as we don't have the progress bar for loading anymore. I'll leave it as is for now, and revisit if we have any issues with this.